src/ActionModal/components/ActionModal.vue

```diff
@@ -442,10 +444,19 @@ export default {
     },
     estimatedFee() {
       // hack
-      return this.networkId.startsWith(`terra`) &&
-        this.selectedBalance.denom !== this.network.stakingDenom
-        ? Number(this.gasPrice) * Number(this.gasEstimate) * Number(this.amount) // already in atoms
-        : Number(this.gasPrice) * Number(this.gasEstimate)
+      // terra uses a tax on all send txs
+      if (
+        this.networkId.startsWith(`terra`) &&
+        this.transactionData.type === transactionTypes.SEND
+      ) {
+        // hardcoding terra tax here until we have it in the API
+        const terraTax = 0.008
+        return (
+          Number(this.gasEstimate) * Number(this.gasPrice) +
+          Number(this.amount) * terraTax
```

Comment:
Oh, now I see what you meant! 

Still, for me it was working without tax... Maybe something only present in mainnet then? I am really curious

Reply:
nope, could reproduce in soju. you set the gas price in two places in the code. that was confusing to me, maybe happened to you as well. reduced to just setting the estimate in one place.